### PR TITLE
30 Add styling to game players tab.

### PR DESCRIPTION
**Pull request by mikelax**
Add styling to game players tab.

To Do List
---
- [x] Change button label / disable button on /game/:id page based on user's status in game
- [x] Pass `game` object in props to JoinGame component
- [ ] Call multiple mutate functions in _submit function in JoinGame. This is needed to insert the message after the game player mutation
- [x] Error handling in mutation?? ie. if player is already pending, they should not be able to join again.

### CODE REVIEW

File: client/src/components/Games/View/components/assets/GamePlayers.styl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.game-player
+  display grid
```
Comment (nazar) [inline client/src/components/Games/View/components/assets/GamePlayers.styl:2]: ❤️ 👍

**Review by nazar — APPROVED** [review]
💯

File: client/src/components/Games/View/components/assets/GamePlayers.styl (hunk 1 above)
Comment (mikelax) [reply client/src/components/Games/View/components/assets/GamePlayers.styl:2]: haha, I think you might like that. 
I was wondering @nazar  if you could take 10 minutes to try and improve the layout between the profile image, name, and to be sure the "status" column stays lined up. 

then the only thing we would need to refactor in the future is once we add support for the Character, which would be either a "Pending", N/A (for game master), or the actual character (image and name) selected.

Comment (nazar) [reply client/src/components/Games/View/components/assets/GamePlayers.styl:2]: I'm currently giving semantic-ui a twirl. This example looks relevant: https://react.semantic-ui.com/elements/list#list-example-image

**Comment by mikelax** [comment]
@nazar  I updated this PR with some basic support for the /join page. 
There are a few concepts that I am not sure how to implement right now, they are the checkboxes in this PR. The main question is how to get props to the JoinGame component (without having the page make a graphql call). Also how to get the _submit function to call multiple mutations. Also need to know if refretchQueries is required. I *think* it will be as the user's local cache will need to be updated to get the new game player list, unless that will refresh when the user is redirected there.

**Comment by nazar** [comment]
>There are a few concepts that I am not sure how to implement right now

Thanks for the work on this @mikelax - I can have a look at the questions raised and take it from here.

>Also how to get the _submit function to call multiple mutations

In terms of calling multiple mutations, these need to be defined in the `grapql` wrapper, like [here](https://github.com/mikelax/forgingadventures/blob/master/client/src/components/Games/View/components/GameLoungeMessages.js#L225-L227). Multiple mutations can be define like this, each with their own name which can then be [called from the props](https://github.com/mikelax/forgingadventures/blob/master/client/src/components/Games/View/components/GameLoungeMessages.js#L172).

**Comment by mikelax** [comment]
thanks @nazar. I think I need to read up a bit more on thunk, as I saw some of the functions used elsewhere were related to that project. So I am still trying to work out how a page, ie. in this case "JoinGame" would get the info it needs, ie. the `game` object.

**Comment by nazar** [comment]
>So I am still trying to work out how a page, ie. in this case "JoinGame" would get the info it needs, ie. the game object.

I haven't had a chance to have a good 👁  at the PR yet but one way to get the `game` object is to use a [graphQL query](https://github.com/mikelax/forgingadventures/blob/master/client/src/components/Games/View/index.js#L45-L47), like we do with the [View component](https://github.com/mikelax/forgingadventures/blob/master/client/src/components/Games/View/index.js#L55).

**Comment by nazar** [comment]
>without having the page make a graphql call

My understanding of graphQL is it should used in any component that requires data. The graphQL client is clever enough to figure out if the requested object is in the cache or if it needs to make a call to the API server. We can also use the `recompose` library to add multiple queries, mutations, etc to a component like [here](https://github.com/mikelax/forgingadventures/blob/master/client/src/components/Games/View/index.js#L44-L50).

For the JoinGame component, [withRouter](https://github.com/mikelax/forgingadventures/blob/master/client/src/components/Games/List/index.js#L97) can also be added to the `compose` function to supply routing props (i.e. `match` to extract the `:id`). Having said that, the JoinGame component should already have routing props as it has been defined as a route Component [here](https://github.com/mikelax/forgingadventures/blob/30-joingame2/client/src/components/Games/index.js#L21).

**Comment by nazar** [comment]
Hey @mikelax - please feel free to assign this ticket over to me once you've had enough :D

**Comment by mikelax** [comment]
haha, @nazar  if you are going to take a peak at it tonight feel free to grab it now. 
I was going to try to wire up connect to add in the additional graphql query, but you can work on it. and then i'll review the changeset.

**Comment by nazar** [comment]
hehe - I'll most likely have a look tomorrow morning. Amanda came back 30 mins ago and going to catch up with her.

**Comment by mikelax** [comment]
@nazar  I am going to merge this PR in as well. 
We have the main use case covered. We will address changes the textarea to the main editor in a separate PR that refactors it to a RichEditor component.

**Comment by mikelax** [comment]
Addresses #30

**Pull request merged**